src/main/java/com/google/api/codegen/config/FieldConfig.java

```diff
@@ -132,8 +132,9 @@ static FieldConfig createFieldConfig(
         && !messageFieldResourceNameConfig.equals(flattenedFieldResourceNameConfig)) {
       // We support the case of the flattenedField using a specific resource name type when the
       // messageField uses a oneof containing that type
```

Comment:
Maybe update this comment with something like: "We support the case of the flattenedField using a specific resource name type when the messageField uses a oneof containing that type **or when the messageField accepts any resource name.**"